dataset/src/test/scala/frameless/CreateTests.scala

```diff
@@ -124,12 +125,27 @@ class CreateTests extends TypedDatasetSuite with Matchers {
     illTyped("TypedDataset.create(data)", ".*could not find implicit value for parameter encoder.*")
   }
 
-  test("not alligned columns should throw an exception") {
+  test("not aligned columns should throw an exception") {
     val v = Vector(X2(1,2))
     val df = TypedDataset.create(v).dataset.toDF()
 
     a [IllegalStateException] should be thrownBy {
       TypedDataset.createUnsafe[X1[Int]](df).show().run()
     }
   }
+
+  test("dataset with different column order") {
+    // e.g. when loading data from partitioned dataset
+    // the partition columns get appended to the end of the underlying relation
+    def prop[A: Arbitrary: TypedEncoder, B: Arbitrary: TypedEncoder] = forAll {
+      (a1: A, b1: B) => {
+        val ds = TypedDataset.create(
+          Vector((b1, a1))
+        ).dataset.toDF("b", "a").as[X2[A, B]](TypedExpressionEncoder[X2[A, B]])
```

Comment:
Why do you have to pass this `TypedExpressionEncoder` explicitly?

Reply:
Otherwise `DataFrame.as` is going to use spark encoders.